apps/prairielearn/src/sync/course-db.ts

```diff
@@ -673,23 +680,25 @@ export async function loadCourseInfo({
   return loadedData;
 }
 
-async function loadAndValidateJson<T extends { uuid: string }>({
+async function loadAndValidateJson<T extends ZodSchema>({
   coursePath,
   filePath,
   defaults,
   schema,
+  zodSchema,
   validate,
   tolerateMissing,
 }: {
   coursePath: string;
   filePath: string;
   defaults: any;
   schema: any;
+  zodSchema: T;
   /** Whether or not a missing file constitutes an error */
   tolerateMissing?: boolean;
-  validate: (info: T) => Promise<{ warnings: string[]; errors: string[] }>;
-}): Promise<InfoFile<T> | null> {
-  const loadedJson: InfoFile<T> | null = await loadInfoFile({
+  validate: (info: z.input<T>) => Promise<{ warnings: string[]; errors: string[] }>;
```

Comment:
I setup the validation function to run before the zod reparse. So the validation takes the input type, but the function returns the output type.